include/nds/rsa.h

```diff
@@ -14,23 +20,77 @@ extern "C" {
 
 #include <nds/ndstypes.h>
 
+/// Context that holds information about the RSA heap
 typedef struct swiRSAHeapContext
 {
-    void *heapStart;
-    void *heapEnd;
-    size_t heapSize;
+    void *heapStart __attribute__((aligned(4)));    ///< Start of the heap rounded up to 4-byte boundary
+    void *heapEnd __attribute__((aligned(4)));      ///< start + size rounded down to 4-byte boundary
+    u32 heapSize;    ///< Heap size matched to the above rounded values
 } swiRSAHeapContext_t;
 
+/// Struture that holds pointers to the RSA buffers
 typedef struct swiRSAbuffers
 {
-    void *dst;
-    const void *sig;
-    const void *key;
+    void *dst;  ///< Pointer to the output buffer
+    const void *sig;    ///< Pointer to the signature buffer
+    const void *key;    ///< Pointer to the RSA key buffer
 } swiRSAbuffers_t;
 
-int swiRSAInitHeap(swiRSAHeapContext_t *ctx, void *heapStart, size_t heapSize);
-int swiRSADecryptRAW(swiRSAHeapContext_t *ctx, swiRSAbuffers_t *rsabuffers, size_t len);
+/// Initialize RSA Heap
+///
+/// @param ctx 12-byte heap information structure that gets set with heap
+///            start, heap end and heap length.
+/// @param heapStart Pointer to the start of the heap.
+/// @param heapSize Size of the heap in bytes (should be usually 4096).
+void swiRSAInitHeap(swiRSAHeapContext_t *ctx, void *heapStart, u32 heapSize);
```

Comment:
Some suggestions here:

1. `4096` could probably be extracted into a constant, like `#define SWI_RSA_DEFAULT_HEAP_SIZE 4096`.
2. `swiRSAInitHeap`, to me, clearly exists to be plugged into an external allocator - which we do have, in the form of picolibc's `malloc()`. I would like to see a wrapper `swiRSAInit` which only takes a `heapSize` argument and handles allocation automatically (as well as a matching `swiRSAExit` which performs deallocation); in this variant, to further make things friendly, a `heapSize` of `0` would be treated as `SWI_RSA_DEFAULT_HEAP_SIZE`.

@AntonioND  ?

Reply:
1. Yeah, good idea.
2. We can add the wrapper later (once we have an example that actually works). We have a similar situation with the decompression routines, where we have the raw SWI calls and then wrapper functions around them.